src/NativeScript/Calling/FFICall.cpp

```diff
@@ -57,17 +64,34 @@ void FFICall::initializeFFI(VM& vm, const InvocationHooks& hooks, JSCell* return
     this->_stackSize += malloc_good_size(sizeof(void * [this->_cif->nargs]));
 
     this->_returnOffset = this->_stackSize;
-    this->_stackSize += malloc_good_size(std::max(this->_cif->rtype->size, sizeof(ffi_arg)));
+    this->_stackSize += malloc_good_size(std::max(this->_cif.get()->rtype->size, sizeof(ffi_arg)));
 
     for (size_t i = 0; i < this->_argsCount; i++) {
         this->_argValueOffsets.push_back(this->_stackSize);
         this->_stackSize += malloc_good_size(std::max(this->_cif->arg_types[i]->size, sizeof(ffi_arg)));
     }
 }
 
+std::shared_ptr<ffi_cif> FFICall::checkForExistingCif(unsigned int nargs, ffi_type* rtype, ffi_type** atypes) {
+
+    std::unordered_map<std::vector<const ffi_type*>, std::shared_ptr<ffi_cif>, SignatureHash>::const_iterator it = FFICache::global()->cifCache.find(this->signatureVector);
+
+    if (it == FFICache::global()->cifCache.end()) {
+        std::shared_ptr<ffi_cif> shared(new ffi_cif, deleteCif);
```

Comment:
I find it a bit confusing that a method called `checkForExistingCif` actually creates a new `cif` in case there is no existing one. I suggest renaming the method (e.g. `getCif` or something) or moving the `cif` creation logic outside the method.
However, this is more or less a subjective statement, so if you don't agree, I am perfectly fine to leave it as is.